src/runtime/vm/vm.cc

```diff
@@ -567,7 +590,7 @@ void VirtualMachine::RunLoop() {
         auto caller_return_register = frames_.back().caller_return_register;
 
         if (PopFrame() == frame_start) {
-          return;
+          goto vm_completed;
```

Comment:
i think you could just simply move the sentence of `vm_completed` here and could avoid `goto`.

Reply:
Makes sense, I'll update it later.